File: python/html_converter.py
"""
This is a variant on Luca Minudel's original exercise created by Emily Bache
"""
import html as html_converter


class HtmlConverter:

    def convert_to_html(self, filename):
        with open(filename, "r") as f:
            # bug: should be "<html>"
            html = "<html<"
            for line in f:
                html += "<p>"
                line = line.strip()
                html += self.escape(line, quote=True)
                # bug: should be "</p>"
                html += "<br />"
            html += "</html>"
            return html

    def escape(self, line, quote):
        return html_converter.escape(line, quote=quote)


if __name__ == "__main__":
    import sys
    if len(sys.argv) < 2:
        print("Usage: html_converter.py <filename>")
        sys.exit(1)
    filename = sys.argv[1]
    converter = HtmlConverter()
    print(converter.convert_to_html(filename))

File: cpp/src/HtmlConverter.cpp


#include "HtmlConverter.h"

#include <iostream>

std::string HtmlConverter::convertToHtml(std::string filename) {
    std::string html = "";
    std::ifstream file(filename);
    if (file.is_open()) {
        // bug: should be "<html>"
        html += "<html<";
        std::string line;
        while (getline(file, line)) {
            html += "<p>";
            line = StringEscapeUtils::trim(line);
            html += StringEscapeUtils::escapeHtml(line);
            // bug: should be "</p>"
            html += "<br />";
        }
        html += "</html>";
    
        file.close();
    } else {
        std::cerr << "Unable to open file: " << filename << std::endl;
    }
    
    return html;
}
